syntax = "proto3";

import "buf/validate/validate.proto";

service RichTextService {
  rpc RenderMarkdown(RenderMarkdownRequest) returns (RenderMarkdownResponse) {
    option idempotency_level = NO_SIDE_EFFECTS;
  }
}

message RenderMarkdownRequest {
  string text = 1 [(buf.validate.field).string = {max_len: 50000}];
}

message RenderMarkdownResponse {
  string html = 1;
}
